android/src/main/java/org/mozilla/mozstumbler/client/Updater.java

```diff
@@ -40,6 +42,12 @@ public boolean wifiExclusiveAndUnavailable(Context c) {
         return !NetworkInfo.getInstance().isWifiAvailable() && ClientPrefs.getInstance(c).getUseWifiOnly();
     }
 
+    public boolean checkForUpdatesRateLimited(final Activity activity, String api_key) {
+        if (System.currentTimeMillis() - sLastUpdateCheck < UPDATE_CHECK_FREQ) {
+            return false;
+        }
+        return this.checkForUpdates(activity, api_key);
```

Comment:
In fact, we should probably just inline the code in `checkForUpdatesRateLimited` into `checkForUpdates`.  We already have a test class over here:

https://github.com/mozilla/MozStumbler/blob/dev/android/src/test/java/org/mozilla/mozstumbler/client/UpdaterTest.java#L46

You should be able to just add instrumentation of the MockSystemClock and verify that the update hasn't actually occured.